### ISSUE TITLE
how to recompile ir file contains something like __asm_movi back to binary?

### ISSUE THREAD

**Issue by miz2019** [avast/retdec#847]
When I lift a binary to llvmir, I find for example
`0x9e4c10:    00 e4 00 6f        movi v0.2d, #0000000000000000`
the inst is correctly decoded, but the llvmir is 
`%5 = call i128 @__asm_movi(i128 %3, i64 0), !insn.addr !25 `
the inst becomes a call. 
according to https://retdec.com/static/publications/retdec-slides-botconf-2017.pdf page 25-26.  This is what retdec designed as. 
'full semantics only for simple instuctions'

The question is how to compile it back to binary given the movi is a call now?

Thanks.

**Comment by silverbacknet** [comment]
Retdec output is almost never directly recompilable; it's made for analysis, not for recompilation. But in this case I think you should see __asm_movi as an intrinsic that should inline down into an actual single movi instruction, even if it's shown as a call. Having a return value is required in ll because most instructions have side effects that have to be captured as well. (I don't know if movi in particular does. I'm not very familiar with how arm transmits exceptions.)

**Comment by miz2019** [comment]
Yes, that requires maybe a codegen/mir pass in recompiling, I wonder that is done by someone else somewhere already.
